.flake8

```diff
@@ -0,0 +1,4 @@
+[flake8]
+max-line-length = 120
+exclude = .venv, venv
+ignore = D101, D102, D103, D107
```

Comment:
we should create some tasks in the future to address this exclusions, but I thought this is good enough for now

Reply:
This looks pretty good to me - it would be nice to add class-level docstrings, but in every function I find they get pretty noisy and hard to maintain. It's always possible to add them in more complex functions without requiring them everywhere